### ISSUE TITLE
Vertex colors are assigned incorrectly in quadrilaterals drawn using CommandPrimitive

### ISSUE THREAD

**Issue by Lielay9** [godotengine/godot#97593]
### Tested versions
Master as of writing: v4.4.dev.custom_build [76a135926]
Regression after #92797 (bisected)

### System information

Godot v4.4.dev (a657ea42f) - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4090 (NVIDIA; 31.0.15.3699) - AMD Ryzen 9 7950X 16-Core Processor (32 Threads)

### Issue description

Quadrilaterals drawn using CommandPrimitive (`RenderingServer.canvas_item_add_primitive()`, antialiasing etc.) assign the provided vertex colors incorrectly after #92797 in the RD-renderers. More specifically, the second constructed triangle (top left, bottom right, bottom left in the first image) uses the colors indexed at 0, 1 and 2, when it should use 0, 2 and 3.
```gdscript
func _draw() -> void:
	var uvs: PackedVector2Array = [Vector2.ZERO, Vector2.RIGHT, Vector2.ONE, Vector2.DOWN]
	var colors: PackedColorArray = [Color.WHITE, Color.RED, Color.GREEN, Color.BLUE]
	var points: PackedVector2Array = uvs.duplicate()
	for i: int in points.size():
		points[i] *= 500
		points[i] += Vector2(50, 50)
	draw_primitive(points, colors, uvs)
```
<img width="566" alt="Screenshot 2024-09-28 204209" src="https://github.com/user-attachments/assets/76107945-df7e-4589-bd39-081e4dba0cb7">

Incorrect antialiasing of line (drawn with scale 10 to blow out the feather size):

<img width="395" alt="Screenshot 2024-09-28 210515" src="https://github.com/user-attachments/assets/2ce6977c-b6e6-4adc-940f-90eeb664e64c">

Compatibility for reference:

<img width="388" alt="Screenshot 2024-09-28 211249" src="https://github.com/user-attachments/assets/041b96b3-ebea-4e89-8997-440fe6e33867">

### Steps to reproduce

↕

### Minimal reproduction project (MRP)

[primitivevertexcolors.zip](https://github.com/user-attachments/files/17175703/primitivevertexcolors.zip)

**Comment by Lielay9** [comment]
And unlike https://github.com/godotengine/godot/issues/97586, this one isn't fixed by #97340.

**Comment by stuartcarnie** [comment]
Reproduced it – thanks for the MRP!

Fixed in #97340:

![CleanShot 2024-09-29 at 06 31 50@2x](https://github.com/user-attachments/assets/168ebed3-31dd-42ac-9d4a-c759da94003d)
